src/network/tarcap/new_taraxa_capability.cpp

```diff
@@ -121,26 +97,37 @@ TaraxaCapability::TaraxaCapability(std::weak_ptr<dev::p2p::Host> _host, NetworkC
   // Creates periodic events
   const auto lambda_ms_min = pbft_mgr ? pbft_mgr->getPbftInitialLambda() : 2000;
 
-  // TODO: use sendTransactions() from TransactionPacketHandler when ready
-  //  if (conf.network_transaction_interval > 0) {
-  //    periodic_events_tp_.post_loop( { conf.network_transaction_interval }, [this] { sendTransactions(); });
-  //  }
-
-  const auto &handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::StatusPacket);
-  auto status_packet_handler = std::static_pointer_cast<StatusPacketHandler>(handler);
+  // Send new txs periodic event
+  const auto &tx_handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::TransactionPacket);
+  auto tx_packet_handler = std::static_pointer_cast<TransactionPacketHandler>(tx_handler);
+  if (conf.network_transaction_interval > 0) {
+    periodic_events_tp_.post_loop(
+        {conf.network_transaction_interval}, [tx_packet_handler = std::move(tx_packet_handler), trx_mgr] {
+          tx_packet_handler->onNewTransactions(trx_mgr->getNewVerifiedTrxSnapShotSerialized(), false);
+        });
+  }
 
-  const auto check_alive_interval = 6 * lambda_ms_min;
-  periodic_events_tp_.post_loop({check_alive_interval},
-                                [this, status_packet_handler] { status_packet_handler->checkLiveness(); });
+  // Check liveness periodic event
+  const auto &status_handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::StatusPacket);
+  auto status_packet_handler = std::static_pointer_cast<StatusPacketHandler>(status_handler);
+  const auto check_liveness_interval = 6 * lambda_ms_min;
+  periodic_events_tp_.post_loop({check_liveness_interval}, [status_packet_handler = std::move(status_packet_handler)] {
+    status_packet_handler->checkLiveness();
+  });
 
+  // Logs packets stats periodic event
   if (conf.network_performance_log_interval > 0) {
     periodic_events_tp_.post_loop({conf.network_performance_log_interval},
-                                  [this] { peers_state_->packets_stats_.logStats(); });
+                                  [peers_state = peers_state_] { peers_state->packets_stats_.logStats(); });
   }
 
-  // TODO: implement logNodeStats()
-  //  const auto summary_interval_ms = 5 * 6 * lambda_ms_min;
-  //  periodic_events_tp_.post_loop( { summary_interval_ms }, [this] { logNodeStats(); });
+  // Logs node stats periodic event
+  const auto node_stats_log_interval = 5 * 6 * lambda_ms_min;
+  // node_stats is shared_ptr as it is probably less expensive to move periodically than object with uint64_t members
```

Comment:
this comment is misleading, I am ok with shared_ptr but you are not periodically moving any  "object with uint64_t members" but whole anonymous function and it's capture is evaluated only once

Reply:
so what is this if not recursive loop where you periodically move whole lambda function, which contains object with uint64_t members ? https://github.com/Taraxa-project/taraxa-node/blob/tarcap_summary_and_send_txs/src/util/thread_pool.cpp#L69

Move operator on lambda internally calls move operator on all captured objects

Reply:
`class Test {
  public:
    Test() = default;
    Test(const Test&) { std::cout << "C"; }
    Test(Test&&) { std::cout << "M"; }
};

int main()
{
    Test test;

    std::function<void()> lambda = [test = std::move(test)]() { std::cout << "lambda"; };
    //auto lambda = [test = std::move(test)]() { std::cout << "lambda"; };
    auto lambda1 = std::move(lambda);
    auto lambda2 = std::move(lambda1);
    auto lambda3 = std::move(lambda2);
}`

hmm when used std::function it prints MM, when using auto it prints MMMM, interesting

Reply:
https://web.mst.edu/~nmjxv3/articles/lambdas.html please read
Even
```
std::function<void()> lambda = [test = std::move(test)]() { std::cout << "lambda"; };
```

will print it twice, this of `test` as anonymous class variable

Reply:
thanks for writing me what I just wrote a commet above :D That was a comment that was supposed to say yes you are right std::function does not call move constructor multiple times, auto = []() {} does